docs/handbooks/state_transition.md

```diff
@@ -0,0 +1,197 @@
+# State Transition Handbook
+
+Before delving into the LECC state transition functions implementation, ensure you've covered the following:
+
+- **Complete Reading**: Go through this entire document.
+- **Relevant Documentation**: Familiarize yourself with related documentation. Relevant links are provided in the "References" section.
+- **Comprehension**: Understand the mechanics and purpose of what you're about to implement. Ensure you can explain it to others.
+- **Context**: Grasp why the implementation is necessary and its significance to the project.
+- **Seek Clarifications**: If in doubt, don't hesitate to ask questions in the [Telegram group](https://t.me/lambdaconsensus).
+- **Testing**: Ensure your implementation passes the relevant tests.
+- **Optimization**: Strive for efficiency in your code.
```

Comment:
>- **Optimization**: Strive for efficiency in your code.

We shouldn't mention this because we don't really want efficiency from the get-go (just sensible performance). Once the client is fully working we can do some benchmarks and optimize it.